tests/mocks/createInsightsClient.ts

```diff
@@ -16,22 +24,22 @@ export function createInsights<TVersion extends string | undefined = '2.4.0'>({
       requestFn: jest.fn(),
     })
   );
+  const mockedInsightsClient = jest.fn(
+    getFunctionalInterface(analytics)
+  ) as unknown as jest.Mock<InsightsClient>;
```

Comment:
We had an error in the `__tests__/createInsightsMiddleware.ts` file:
```
Type 'Mock<void, [method: keyof InsightsMethodMap, options?: InitParams | undefined] | [method: keyof InsightsMethodMap, callback: (version: string) => void] | ... 12 more ... | [method: ...]>' is not assignable to type 'ProvidedInsightsClient'.
```
Not sure if there is a better way to fix this.

Reply:
you can use the `castToJestMock` utility function we have that does the same